### PR TITLE
Fix for #1057 Patient Ledger Reports Alert

### LINKED ISSUE #1057: Patient ledger report needs to say no data to display , Currently it just disappears.
The patient ledger report results aren't displayed long enough.
- Fill the form and submit.
Observe the results show up, and disappear almost immediately.

### PR DESCRIPTION
**Pull request by amughalbscs16**
fixes #1057  
Now when there are no records, in the patient ledger. There is an alert showing that there are no records. 
They should search again. 
Here is the screenshot.

no matches on reports patient ledger

### CODE REVIEW

**Comment by Trodrige** [comment]
The message appears in the alert and before the table. Keep one!
Do we need the table head if there's no result to display?

**Comment by amughalbscs16** [comment]
It gives user a better view and idea and hints about the fact that no
records exist. It's about user experience

**Comment by amughalbscs16** [comment]
I appreciate that you are trying to help me but I did that because It gives a better and fast appearance of empty table to the user and a user feels like easier to interpret the meaning. We have studied this thing in our subject Human Computer Interaction.

**Comment by aethelwulffe** [comment]
Let's not go against the HCI instructor!  😈 

If you want to tweak this, you could always say 
echo GLOBALS[whatever the clinic name value is]." EHR says: you didn't find nuthin'"  Instead of echoing the host name...which could be very weird and frightening to some non-tech ppl.  
  It's something I used to teach in my Control Interface classes.  ⚙️ 🙉 😸   Don't print out any technobabble where the "normal" people can see it if you don't want sysadmin support calls in the middle of the night.

**Comment by amughalbscs16** [comment]
So you want that it should remove the word local Host from the alert right?

**Comment by teryhill** [comment]
Can you remove the "Localhost" from the dialog. Just give the message. @amughalbscs16 . I have and am testing the changes.

**Comment by naveen17797** [comment]
@amughalbscs16 use jquery modal instead of alert, it will be user friendly
its just a suggestion :relaxed:

**Comment by amughalbscs16** [comment]
Okay, I'll do that.

**Comment by amughalbscs16** [comment]
I gave that suggestions to Terry Hill you can read in the issue he told me
to tryalert instead so I went with alert.

**Comment by teryhill** [comment]
What does teryhill know. 8^) Try it with the modal and lets see what it looks like.

**Comment by naveen17797** [comment]
screenshot from 2018-03-21 21-37-34

it would like this, when done with jquery ui. design a model like this

**Comment by aethelwulffe** [comment]
alert() is less expensive of course, but I would pay the price to keep 
from freaking out or confusing our mental health counselors...they are 
kinda nuts anyway.

-Sorry for the noise over little nit-pickies.  Few people actually enjoy 
the actions of publishing editors...and every one of us needs to play 
that role for each other.

**Comment by amughalbscs16** [comment]
I'm on it

**Comment by aethelwulffe** [comment]
Commit that!

**Comment by amughalbscs16** [comment]
Same has been setup for Admin Patient Ledger now you can merge the commit @aethelwulffe  Thanks for the guidance.

File: interface/reports/pat_ledger.php [hunk 1]
```diff
@@ -347,20 +347,72 @@ function sel_patient() {
...
   }
+  .modal-content {
+      background-color: #fefefe;
+      margin: 15% auto; /* 15% from the top and centered */
+      padding: 20px;
+      border: 1px solid #888;
+      width: 80%; /* Could be more or less, depending on screen size */
+  }
+
+  /* The Close Button */
```
Comment (aethelwulffe) [inline interface/reports/pat_ledger.php]: I like the documentation here 🥇

File: interface/reports/pat_ledger.php [hunk 2]
```diff
@@ -347,20 +347,72 @@ function sel_patient() {
...
 
 <title><?php echo xlt('Patient Ledger by Date') ?></title>
 
 <script language="JavaScript">
+  
  $(document).ready(function() {
   var win = top.printLogSetup ? top : opener.top;
   win.printLogSetup(document.getElementById('printbutton'));
  });
+function hideNoRecordsModal(){
```
Comment (aethelwulffe) [inline interface/reports/pat_ledger.php]: Another function for the report generator system to grab.

File: patient_portal/report/pat_ledger.php [hunk 3]
```diff
@@ -436,10 +458,38 @@ function sel_patient() {
       <?php // can add any additional javascript settings to datetimepicker here; need to prepend first setting with a comma ?>
  });
   });
+  function showNoRecordsModal(){
+    var modal = document.getElementById('noLedgerRecords');
```
Comment (aethelwulffe) [inline patient_portal/report/pat_ledger.php]: Be good in an public library.  I am just taking note @Trodrige

**Pull request merged**